src/main/kotlin/no/nav/familie/ba/sak/infotrygd/MigreringService.kt

```diff
@@ -0,0 +1,218 @@
+                if (vilkår.contains(it.vilkårType)) {
+                    it.resultat = Resultat.OPPFYLT
+                    it.begrunnelse = "Migrering"
+                }
+            }
+        }
+    }
+
+    private fun Vilkårsvurdering.forsøkSettPerioderFomTilpassetInfotrygdKjøreplan() {
+        val inneværendeMåned = YearMonth.now()
+        this.personResultater.forEach { personResultat ->
+            personResultat.vilkårResultater.forEach {
+                it.periodeFom = virkningsdatoFra(infotrygdKjøredato(inneværendeMåned))
+            }
+        }
+    }
+
+    private fun virkningsdatoFra(kjøredato: LocalDate): LocalDate {
+        LocalDate.now().run {
+            return when {
+                this.isBefore(kjøredato) -> this.førsteDagIInneværendeMåned()
+                this.isAfter(kjøredato.plusDays(1)) -> this.førsteDagINesteMåned()
+                env.erDev() || env.erE2E() -> this.førsteDagINesteMåned()
+                else -> throw FunksjonellFeil("Migrering er midlertidig deaktivert frem til ${kjøredato.plusDays(2)}",
+                                              "Migrering er midlertidig deaktivert frem til ${kjøredato.plusDays(2)}")
+            }
+        }
+    }
+
+    private fun infotrygdKjøredato(yearMonth: YearMonth): LocalDate {
+        yearMonth.run {
+            if (this.year != 2021) throw Feil("Migrering feilet: Kopien av Infotrygds kjøreplan er utdatert.",
+                                              "Migrering feilet: Kopien av Infotrygds kjøreplan er utdatert.",
+                                              httpStatus = HttpStatus.INTERNAL_SERVER_ERROR)
+            return when (this.month) {
+                APRIL -> 19
+                MAY -> 14
+                JUNE -> 17
+                JULY -> 19
+                AUGUST -> 18
+                SEPTEMBER -> 17
+                OCTOBER -> 18
+                NOVEMBER -> 17
+                DECEMBER -> 6
+                else -> 19
+            }.run { yearMonth.atDay(this) }
+        }
+    }
+
+    private fun sammenlignTilkjentYtelseMedBeløpFraInfotrygd(behandling: Behandling,
+                                                             løpendeSak: Sak) {
+        tilkjentYtelseRepository.findByBehandlingOptional(behandling.id)?.andelerTilkjentYtelse?.let {
+            if (it.isEmpty()) throw Feil("Migrering feilet: Fant ingen andeler tilkjent ytelse på behandlingen",
+                                         "Migrering feilet: Fant ingen andeler tilkjent ytelse på behandlingen",
+                                         HttpStatus.INTERNAL_SERVER_ERROR)
+
+            val førsteUtbetalingsperiode = beregnUtbetalingsperioderUtenKlassifisering(it)
+                    .sortedWith(compareBy<LocalDateSegment<Int>>({ it.fom }, { it.value }, { it.tom }))
+                    .first()
+            val tilkjentBeløp = førsteUtbetalingsperiode.value
+            val beløpFraInfotrygd = løpendeSak.stønad!!.delytelse.first().beløp.toInt()
```

Comment:
Her kan jeg begynne å lure på om vi velger den første fra datosortert delytelse, men egentlig er det vel her antatt at det er én? Hvis det er tilfellet, kan man tydeliggjøre det vel å bruke singleOrNull? 
```suggestion
            val beløpFraInfotrygd = løpendeSak.stønad!!.delytelse.singleOrNull()?.beløp?.toInt() ?: error("Finnes flere delytelser på sak")
```

Reply:
👍